wordpress-mobile/WordPress-Android
# Offline Sync: EditPostActivity Conversion

Fixes #20509 

This PR marks the initial phase in simplifying the complexity of `EditPostActivity`.

In this PR is the conversion of `EditPostActivity` from Java to Kotlin. While Android Studio handled the bulk of the conversion, I manually did a bunch of cleanup to minimize the need for "most" suppressions. The lingering suppressions, include `CyclomaticComplexMethod`, `ReturnCount`, `LargeMethod`, and `LargeClass`. They will be addressed outside of this PR.

The ultimate objective is to refactor this class into coherent segments, streamlining the activity into a more manageable and maintainable structure. 

The UI was not touched during this conversion

-----

## To Test:
Using the branch from this PR, thoroughly check the following create/edit post scenarios (and any others you can think of)
- ✅ Verify you can create a post
- ✅ Verify you can edit a post
- ✅ Verify you can edit post setting from within the edit post view
- ✅ Verify you can edit change the publish post settings
- ✅ Verify you can create/edit a post with images, galleries
- ✅ Verify you can create/edit a post with videos
- ✅ Verify your work is auto saved (use the web to look for changes)
- ✅ Verify you can return from background
- ✅ Verify you can save a post as a draft
- ✅ Verify you can't publish a post when media is uploading
- ✅ Verify you can set a featured image

- ✅ Verify CI passed the unit tests
- ✅ Verify CI passed the instrumentation tests
-----

## Regression Notes

1. Potential unintended areas of impact
Edit post activities no longer work

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing of edit posts. No unit tests exist because it's an activity

3. What automated tests I added (or what prevented me from doing so)
The existing instrumentation tests.

-----

## PR Submission Checklist:

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
